### PR TITLE
Spec for MSC2867 (Marking rooms as unread)

### PR DESCRIPTION
**Pull request by Johennes**
Per [MSC2867](https://github.com/matrix-org/matrix-spec-proposals/pull/2867) ([markdown](https://github.com/Bubu/matrix-doc/tree/unread_rooms/proposals/2867-rooms_marked_unread.md))

Relates to: matrix-org/matrix-spec-proposals#2867
Signed-off-by: Johannes Marbach <n0-0ne+github@mailbox.org>

### Pull Request Checklist

* [x] Pull request includes a [changelog file](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#adding-to-the-changelog)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#sign-off)
* [ ] Pull request is classified as ['other changes'](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#other-changes)

Preview: https://pr1895--matrix-spec-previews.netlify.app

### CODE REVIEW

**Comment by Johennes** [comment]
This currently generalizes the "Fully read markers" module to include read and unread markers because there are some shared semantics between these two. I'm curious if folks think this is a good idea. If not, the added subsections are easily moved to their own module.

Unlike for read markers, the MSC doesn't specify any API for unread markers. I assume this is because for unread markers there is no use case for performing any other actions (such as setting read receipts) simultaneously. Consequently, I haven't added any new CS endpoints here.

The MSC goes into quite a bit of detail around the UI treatment of unread rooms with red badges. I have interpreted this as an example and left the concrete UI treatment as an implementation detail in the spec.

**Review by sumnerevans — APPROVED** [review]

File: content/client-server-api/modules/read_markers.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
+### (Un)read markers
```
[the remark this answers is not in the document]
Comment (Johennes) [reply content/client-server-api/modules/read_markers.md]: Good point. Have made it so.

For the casing in other places, yeah, it triggers my OCD, too. 😅

I can just fix the "Event Annotations and _r_eactions" heading in this PR if that would be ok. The casing in headings seems to be fairly mixed throughout the spec though. Proper title casing is apparently also almost a [research topic](https://titlecaseconverter.com/rules/). 🙈

Screenshot 2024-07-10 at 10 12 47

[the remark this answers is not in the document]
Comment (turt2live) [reply content/client-server-api/modules/read_markers.md]: We use sentence case, fwiw: https://github.com/matrix-org/matrix-spec/blob/main/meta/documentation_style.rst#correct-capitalisation-for-long-section-names

[the remark this answers is not in the document]
Comment (Johennes) [reply content/client-server-api/modules/read_markers.md]: Oh. I had missed that. 😳

Have turned the heading into sentence case now.

**Review by anoadragon453 — APPROVED** [review]

**Comment by anoadragon453** [comment]
Thank you!

**Pull request merged**